test/cluster/basic_test.go

```diff
@@ -466,6 +466,179 @@ func TestNoMajorityToChooseLeader(t *testing.T) {
 	}, 30*time.Second, 100*time.Millisecond)
 }
 
+//Scenario:
+// round 1:
+// - start 3 servers in a cluster.
+// - wait for one to be the leader.
+// - submit a tx.
+// - stop the leader.
+// round 2:
+// - find leader2 and stop it.
+// - submit a tx to the remaining server.
+// - there is no majority to pick leader => tx fails.
+// - restart one server so now there will be a majority to pick a leader.
+// round 3:
+// - find leader3.
+// - submit a tx => tx accepted.
+// - restart the third server => all 3 nodes are active.
+// round 4:
+// - find leader4.
+// - submit a tx => tx accepted.
+func TestStopLeaderNoMajorityToChooseLeader(t *testing.T) {
```

Comment:
This one and `TestStopFollowerNoMajorityToChooseLeader` are almost identical except the last part. Is it possible to extract the common parts into functions?